presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -87,75 +64,148 @@ public CassandraClusteringPredicatesExtractor(List<CassandraColumnHandle> cluste
         return TupleDomain.withColumnDomains(notPushedDown);
     }
 
-    private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColumnHandle> clusteringColumns, TupleDomain<ColumnHandle> predicates)
+    private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColumnHandle> clusteringColumns,
+            TupleDomain<ColumnHandle> predicates, int protocolVersion)
     {
         ImmutableMap.Builder<ColumnHandle, Domain> domainsBuilder = ImmutableMap.builder();
-        ImmutableList.Builder<Set<Object>> clusteringColumnValues = ImmutableList.builder();
+        ImmutableList.Builder<String> clusteringColumnSql = ImmutableList.builder();
         for (CassandraColumnHandle columnHandle : clusteringColumns) {
             Domain domain = predicates.getDomains().get().get(columnHandle);
 
             if (domain == null) {
                 break;
             }
-
-            if (domain.isNullAllowed()) {
-                return new ClusteringPushDownResult(domainsBuilder.build(), ImmutableSet.of());
-            }
```

Comment:
Changes Done. I am breaking the loop now here because I still want the keys to be pushed down till possible.